core/src/main/scala/org/apache/spark/sql/delta/storage/DelegatingLogStore.scala

```diff
@@ -153,6 +154,14 @@ class DelegatingLogStore(hadoopConf: Configuration)
 
 object DelegatingLogStore {
 
+  try {
+    // load any arbitrary delta-storage class to ensure the dependency has been included
+    classOf[io.delta.storage.S3SingleDriverLogStore]
+  } catch {
+    case e: NoClassDefFoundError =>
```

Comment:
I don't think we've seen this issue.

Anyways, `ClassNotFoundException` would not happen here. It would happen [here](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/storage/LogStore.scala#L265).

That could happen if you do
```
.config("spark.delta.logStore.class", "io.delta.storage.FooLogStore")
```

But then the error message seems good enough, no? All we could really add is "Make sure you have entered the right log store class name". This seems obvious though.

What do you think? @vkorukanti 

```
java.lang.ClassNotFoundException: io.delta.storage.FooLogStore
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.spark.util.Utils$.classForName(Utils.scala:216)
	at org.apache.spark.sql.delta.storage.LogStore$.createLogStoreWithClassName(LogStore.scala:263)
	at org.apache.spark.sql.delta.storage.LogStoreProvider.createLogStore(LogStore.scala:295)
```

Reply:
sounds good!

Reply:
nit: why not use the more general interface `io.delta.storage.LogStore`?